Source/Redemption/Private/PowerUpComponent.cpp

```diff
@@ -32,3 +36,51 @@ void UPowerUpComponent::TickComponent(float DeltaTime, ELevelTick TickType, FAct
 	// ...
 }
 
+void UPowerUpComponent::UsePowerUp(const FInputActionValue& Value)
+{
+	UE_LOG(LogTemp, Warning, TEXT("Using Power"));
+	
+	if (CurrentPowerUp == EActivePowerUp::BoneThrow)
+	{
+		UE_LOG(LogTemp, Warning, TEXT("BONE!!!!!!"))
+	}
+	else if (CurrentPowerUp == EActivePowerUp::Hover)
+	{
+		UE_LOG(LogTemp, Warning, TEXT("HOVER!!!!!!"))
+	}
+	else if (CurrentPowerUp == EActivePowerUp::Parry)
+	{
+		UE_LOG(LogTemp, Warning, TEXT("PARRY!!!!!!"))
+	}
+}
+
+void UPowerUpComponent::ChangeMappingContext()
+{
+	if (UEnhancedInputLocalPlayerSubsystem* Subsystem = GET_ENHANCED_INPUT_LOCAL_PLAYER_SUBSYSTEM)
+	{
+		Subsystem->RemoveMappingContext(BoneMappingContext);
+		Subsystem->RemoveMappingContext(HoverMappingContext);
+		Subsystem->RemoveMappingContext(ParryMappingContext);
+
+		if (CurrentPowerUp == EActivePowerUp::BoneThrow)
+		{
+			Subsystem->AddMappingContext(BoneMappingContext, 0);
+		}
+
+		if (CurrentPowerUp == EActivePowerUp::Hover)
+		{
+			Subsystem->AddMappingContext(HoverMappingContext, 0);
+		}
+
+		if (CurrentPowerUp == EActivePowerUp::Parry)
+		{
+			Subsystem->AddMappingContext(ParryMappingContext, 0);
+		}
```

Comment:
can the current power up be more than one? should it be a switch or chain of else if's instead to reduce the checks?

Reply:
it cannot. yes a switch would be better there